isotovideo

```diff
@@ -171,22 +171,13 @@ sub stop_commands {
         my $timeout = 15;
         my $ua      = Mojo::UserAgent->new(request_timeout => $timeout);
         my $tx      = $ua->post($url, json => {stopping_test_execution => $reason});
-        try {
-            my $res = $tx->result;
-            if ($res->code != 200) {
-                my $error_message = $res->to_string;
-                diag('isotovideo: unable to inform websocket clients about stopping command server: ' . $error_message);
-                if ($error_message =~ qr/.*timeout.*/i) {
-                    diag("isotovideo: The command server could be under heavy load and the timeout of $timeout seconds has been exceeded. This message is most likely not be related to the MAX_JOB_TIME of the current job possibly being exceeded.");
-                }
-                else {
-                    diag('The command server might have already been stopped by the worker after the user has aborted the job or the job timeout has been exceeded.');
-                }
-            }
+        if (my $err = $tx->error) {
+            my $msg = $err->{code} ? "$err->{code} response: $err->{message}" : "Connection error: $err->{message}";
+            $msg .= "\nisotovideo: The command server could be under heavy load and the timeout of $timeout"
+              . ' seconds has been exceeded. This message is not be related to the MAX_JOB_TIME'
+              . ' of the current job possibly being exceeded.' if $err =~ qr/.*timeout.*/i;
+            diag('isotovideo: unable to inform websocket clients about stopping command server: ' . $msg) if $cmd_srv_process->is_running;
```

Comment:
the check `if $cmd_srv_process->is_running` is probably not enough to prevent all race conditions here, right? I was more thinking of a try…catch that would just ignore the error in case when we are already shutting down. But maybe we can adjust `$cmd_srv_process->is_running` to not return true in case of already shutting down?

Reply:
But how would isotovideo know whether the command server is currently shutting down?
But this seems all too complicated anyways. Maybe the command server can trigger this in its own signal handler. That would be way more elegant. I tried that also when implementing the developer mode initially. Maybe it is worth a second shot.

Reply:
you mean just relying on signals rather than sending RPC?

Reply:
The command server is terminated sooner or later anyways. Either it received SIGTERM/SIGINT out of our control or we stop it ourselves a few lines below. If the command server would send these web socket messages itself from its own signal handler we would not have to deal with that at all in isotovideo.